src/main/java/ejercicio/banco/service/PaymentServiceImpl.java

```diff
@@ -30,8 +30,11 @@ public PaymentServiceImpl(PaymentRepository repository) {
             // If our bank is involved in payment, resolves it
             if (bestBank.getId() == payments.get(i).getBankId()) {
 
-                // Modifies the money from the user after payment(assuming user is paying)
-                bestBank.getUsers().get(payments.get(i).getUserId() - 1).setMoney(bestBank.getUsers().get(payments.get(i).getUserId() - 1).getMoney() - payments.get(i).getAmount());
+                try {
+                    // Modifies the money from the user after payment(assuming user is paying)
+                    bestBank.getUsers().get(payments.get(i).getUserId() - 1).setMoney(bestBank.getUsers().get(payments.get(i).getUserId() - 1).getMoney() - payments.get(i).getAmount());
```

Comment:
This line is a bit difficult to understand (I know it was here before). Could you refactor it a bit? At least, extracting some variables out of this objects orgy :D